content/components/selectpanel.mdx

```diff
@@ -0,0 +1,222 @@
+</Caption>
+
+### Additional control slot
+
+<img
+  width="960"
+  alt="Three cropped Select panel footers each displaying a different option of a control in the bottom right left corner: Link, button, or checkbox input."
+  src="https://user-images.githubusercontent.com/40274682/228338168-a9424773-531e-41e3-8d95-261eb971c340.png"
+/>
+<Caption>
+  An optional control slot is available in the Select panel’s bottom left corner. Avoid use if not absolutely necessary
+  in order to avoid component and decision overwhelm.
+</Caption>
+
+### Anchoring
+
+Select panel can be anchored to the trigger, or centered within the page. If it’s centered within the page it should have a backdrop.
+
+<Box display="grid" gridTemplateColumns={['1fr', null, null, null, '1fr 1fr']} gridGap={5}>
+  <div>
+    <img
+      src="https://user-images.githubusercontent.com/40274682/228386690-dd175879-0dd6-4798-8fb5-b0ee096eceea.png"
+      alt=""
+    />
+    <Caption>Anchored Select panel to a trigger.</Caption>
+  </div>
+  <div>
+    <img
+      src="https://user-images.githubusercontent.com/40274682/228386689-ea191d76-3a50-41d9-98f0-9dc997caf4bf.png"
+      alt=""
+    />
+    <Caption>Select panel centered with a darker backdrop.</Caption>
+  </div>
+</Box>
+
+## Best practices
+
+### Be predictable
+
+<DoDontContainer>
+  <Do>
+    <img
+      width="464"
+      alt="Select panel with a back button, close button, Heading, subheading, five multiple choice options with three of them selected, and apply and cancel buttons."
+      src="https://user-images.githubusercontent.com/40274682/228386683-20e7ee4e-0d1b-4b6c-9dd1-eddd277b5512.png"
+    />
+    <Caption>If there are more than single options in a menu, have a user confirm their choices.</Caption>
+  </Do>
+  <Dont>
+    <img
+      width="464"
+      alt=""
+      src="https://user-images.githubusercontent.com/40274682/228386685-4699374e-5f98-4dd4-bab8-f488ad0fff8a.png"
+    />
+    <Caption>
+      Never force users to submit their multi-select options by closing the dialog or auto-updating their choices upon
+      selection.
+    </Caption>
+  </Dont>
+</DoDontContainer>
+
+### Click-to-dismiss behavior
+
+<DoDontContainer>
+  <Do>
+    <img
+      width="464"
+      alt="Mouse cursor indicating a clicking action on the apply button."
+      src="https://user-images.githubusercontent.com/40274682/228388751-12c1f46e-aee7-4e03-8899-b273d31f7d1b.png"
+    />
+    <Caption>
+      Guide users to save or cancel their selections to dismiss Select panel to prevent confusion and data loss.
+    </Caption>
+  </Do>
+  <Dont>
+    <img
+      width="464"
+      alt="Mouse cursor indicating a click action outside of the dialog."
+      src="https://user-images.githubusercontent.com/40274682/228388749-445619e6-91fa-4f9f-b300-6e1fe21c62a5.png"
+    />
+    <Caption>
+      Don’t allow a click-outside-to-dismiss behavior for Select panel. Users may be used to closing and submitting the
```

Comment:
The escape key should always close and cancel changes, it should not save or retain any data. Places where this is in practice today is incorrect behavior. 

We can leave in the click to dismiss, but it _**cannot**_ submit any information. Whether or not it retains the information is potentially up for debate, I'd like to check with @jscholes on this. 

As for not seeing who is selected, labels and assignees that have already been applied should be at the top of the listbox, potentially in its own group. Selected items will not move to the top until it's been submitted and reopened.

Reply:
> As for not seeing who is selected, labels and assignees that have already been applied should be at the top of the listbox, potentially in its own group. Selected items will not move to the top until it's been submitted and reopened.

I wasn't clear with my example. Here's a screenshot of the current issue sidebar:

![A screenshot of the issues sidebar. In the screenshot the following headers are captured: Reviewers, Assignees, Labels. Reviewers has 4 reviewers listed, Assignees has none, and Labels has none. Each label has an icon button in the shape of a gear, which when acted upon will open the select panel respective to that heading.](https://user-images.githubusercontent.com/118266/228803054-3b066e02-e369-4608-850e-7fcbc96b9531.png)

If I click the gear to open the reviewers panel, the rest of the sidebar is occluded. Let's say I click the "Reviewers" gear to assign reviewers, but in that moment I realise I want to look at the Assignees list which will help me inform who I should add to reviewers. The only way to glance at the Assignees list is to dismiss the Reviewers panel, so I can check the Assignees list, then I can re-open to the Reviewers panel.

Keyboard users have common gesture available to them to dismiss floating UI; the `Escape` key. For any floating UI, _so far_, mouse users have a similar gesture, which is to click outside of the bounds of the floating UI. As is currently proposed, SelectPanel will deviate from that norm, and instead rely on the Cancel button, which is a smaller hit target.

> The escape key should always close and cancel changes, it should not save or retain any data. Places where this is in practice today is incorrect behavior.

I understand why it should not _submit_ any data, as it is potentially performing destructive actions. Can we please justify why it should not retain data during the page session. I don't understand why `Escape` would effectively reset the form. Is this a common pattern? Does `Escape` reset forms that are not inside floating UI?

Reply:
> If I click the gear to open the reviewers panel, the rest of the sidebar is occluded. Let's say I click the "Reviewers" gear to assign reviewers, but in that moment I realise I want to look at the Assignees list which will help me inform who I should add to reviewers. The only way to glance at the Assignees list is to dismiss the Reviewers panel, so I can check the Assignees list, then I can re-open to the Reviewers panel.

If this becomes an issue for users with the new select panel, we can look into creating a minimizing or hiding behavior but as it's more complex I have not included this in the MVP which really needs to be published to improve this component that's been in limbo with me advocating for it for 2 years. 

> I don't understand why Escape would effectively reset the form. Is this a common pattern?

Yes this is the expected behavior of the escape key. It's the equivalent of "Get me out I don't want to be here". It should not save any information or changes made. 

We can allow click to dismiss if you feel this is necessary but it will result in data loss for users since GitHub has long since employed a anti-pattern. 

So we can either:
- Disable click-to-dismiss to prevent a loss of data entry
OR
- Enable click-to-dismiss and have a confirmation dialog pop up to prevent unsaved changes being lost (though this feels pretty heavy)